### PR TITLE
Correct roll angle units, issue #28256

### LINKED ISSUE #28256: [Bug]: axes3d.py's _on_move() converts the roll angle to radians, but then passes it to view_init() as if it were still in degrees
### Bug summary

In \lib\mpl_toolkits\mplot3d\axes3d.py, `_on_move()` deals with rotation of 3d axes using the mouse.  In line 1522, the roll angle (in degrees) is converted to radians:
`    roll = np.deg2rad(self.roll)`
This roll in radians is used to calculate a new elevation and azimuth, like so:
```
    delev = -(dy/h)*180*np.cos(roll) + (dx/w)*180*np.sin(roll)
    dazim = -(dy/h)*180*np.sin(roll) - (dx/w)*180*np.cos(roll)
    elev = self.elev + delev
    azim = self.azim + dazim
```
A moment later, the view is updated:
```
    self.view_init(
        elev=elev,
        azim=azim,
        roll=roll,
        vertical_axis=vertical_axis,
        share=True,
    )
```
However, `view_init()` expects its parameters to be in degrees, not radians. As a consequence, the roll now diminishes by a factor pi/180 with every mouse movement. Not intended.

### Code for reproduction

```Python
# Run the surface3d.py example, adding
ax.roll = 45
# It shows the plot, in the intended funny orientation (roll=45)
# Then move the mouse - you will see the orientation jump suddenly (to roll=0)
```

### Actual outcome

The figure orientation has jumped to roll=0, after trying to rotate it only slightly by dragging the mouse:
Figure_rotated

### Expected outcome

The figure is close to its original orientation (before dragging the mouse), at roll=45:
Figure_1

### Additional information

Fix: add a line:
`roll = self.roll`
right after updating `elev` and `azim` (i.e., after line 1526).

### Operating system

All, presumably; but I noticed it on Windows

### Matplotlib Version

3.10.0.dev191+ge5af947d1b.d20240517

### Matplotlib Backend

tkagg

### Python version

Python 3.12.3

### Jupyter version

_No response_

### Installation

pip

### PR DESCRIPTION
**Pull request by MischaMegens2**
In mplot3d.Axes3D._on_move(), pass roll angle to view_init() in degrees (as it expects; not converted to radians, mistakenly).
Resolves #28256.

### CODE REVIEW

**Comment by scottshambaugh** [comment]
Thanks for catching this! Is there a way that you can modify the existing mouse drag tests or add a new one to check for this? Those tests are a bit tricky to write / understand since they spoof mouse movements, so if there isn't an easy way to do that I think I would be fine with skipping it.

**Comment by MischaMegens2** [comment]
> Thanks for catching this! Is there a way that you can modify the existing mouse drag tests or add a new one to check for this? Those tests are a bit tricky to write / understand since they spoof mouse movements, so if there isn't an easy way to do that I think I would be fine with skipping it.

I suppose I could try - but which existing tests did you have in mind? It seems most tests are for `roll=0`, with the exception of `test_bar3d_shaded()`, `test_shared_view()` and `test_format_coord()`, but these are clearly for other purposes than `roll` specifically.
We might add a variation on `test_axes3d_rotated()` with non-zero roll, for starters; and turn `test_pan()` into a `test_rotate()` for the mouse, since the rotation doesn't have much test coverage, apparently. And then add a `test_rotate_with_roll()`. Does that sound like a plan? (Or does it sound like it needs a separate PR? Please advise...)

**Comment by scottshambaugh** [comment]
I was thinking about `test_pan` and `test_toolbar_zoom_pan`, but it looks like we don't actually have an equivalent test for rotating the view. If you want to take a shot at writing a similar test for rotating go for it, but it might be tricky so if you're not comfortable with that then we can merge as-is and I'll open an issue to add those tests.

https://github.com/matplotlib/matplotlib/blob/main/lib/mpl_toolkits/mplot3d/tests/test_axes3d.py#L1769-L1880

**Comment by MischaMegens2** [comment]
I'll give it a try and let you know, it might take me a day or two

File: lib/mpl_toolkits/mplot3d/tests/test_axes3d.py
Comment (scottshambaugh) [inline lib/mpl_toolkits/mplot3d/tests/test_axes3d.py:1098]: We have nonzero roll angles in a few other tests, so I don't think we need an additional one here. (test_shared_view, test_bar3d_shaded)

File: lib/mpl_toolkits/mplot3d/tests/test_axes3d.py
Comment (scottshambaugh) [inline lib/mpl_toolkits/mplot3d/tests/test_axes3d.py:1800]: I think this test looks great! I would suggest that you combine both tests into a single function, since the only code different is the roll angle.

File: lib/mpl_toolkits/mplot3d/tests/test_axes3d.py
Comment (scottshambaugh) [reply lib/mpl_toolkits/mplot3d/tests/test_axes3d.py:1098]: When you remove the test image, you'll have to squash your commits otherwise the "PR Cleanliness" check will complain about an image being added and then removed in the same PR.

Comment (MischaMegens2) [reply lib/mpl_toolkits/mplot3d/tests/test_axes3d.py]: Ah good, I took out the superfluous roll test, combined the mouse rotation tests, and force pushed the result.

**Review by scottshambaugh — APPROVED** [review]
This looks good to me, approved and once another maintainer adds their approval as well it will be merged in.

Thank you and congratulations on your first contribution to matplotlib! We hope to see more of you, and contributions are always welcome!

**Comment by MischaMegens2** [comment]
> [...] We hope to see more of you, and contributions are always welcome!

Well, now that we're at it...
Part of your commit 'Add ability to roll the camera in 3D plots' (#21426, 3 year ago...) reads:
`- Make elev, azim, roll ordering consistent everywhere`
This sounds like a good idea to me. However, I am puzzled by your choice of ordering (`elev, azim`): it is just the opposite of the matlab convention; and when I google:
- `"azimuth, elevation"` -> 291 000 results
- `"elevation, azimuth"`-> 41 500 results

And similarly:
- `"azimuth, elevation, roll"` -> 1380
- `"elevation, azimuth, roll"` -> 188

That is to say, the ordering you chose is not the common one. And if you consider that searching for both yields
- `"azimuth, elevation" and "elevation, azimuth"` -> 37 500 results

(some links are non-committal), then it becomes clear that the `elevation, azimuth` order is really a minority position: (41-37)/(291-37) < 2%.
Moreover, the ordering with elevation first does not correspond to the order in which the 3 rotations take place.

Should we raise an issue to change it?
(You had the good foresight to use named parameters, so it is not a completely breaking change...)

**Review by oscargus — APPROVED** [review]

**Comment by scottshambaugh** [comment]
Elevation and azimuth are applied at the same time, in effect neither comes before the other. 

https://github.com/matplotlib/matplotlib/blob/9765379ce6e7343070e815afc0988874041b98e2/lib/mpl_toolkits/mplot3d/axes3d.py#L1057-L1059

**Comment by scottshambaugh** [comment]
Thinking about it more, I think you’re more right about the rotation order. If thinking about extrinsic rotations, then the rotation order is roll, elev, then azim. If thinking about intrinsic rotations then the rotation order is azim, elev, then roll. The definitions in the docs are still unambiguous so I still think this is fairly marginal to make a change of API, but maybe @oscargus can weigh in.

**Comment by MischaMegens2** [comment]
> Thinking about it more, I think you’re more right about the rotation order if this is conceptualized as Euler angles. 

Well actually, to be precise, mplot3d's `azim, elev, roll` are a kind of Tait-Bryan angles, not classic Euler angles ([there exist 12 kinds of each, not counting the signs...](https://github.com/moble/quaternion/wiki/Euler-angles-are-horrible))

> If thinking about extrinsic rotations, then the rotation order is roll, elev, then azim. If thinking about intrinsic rotations then the rotation order is azim, elev, then roll. 

Ah, yes, now we're talking!
The corresponding quaternion (for the intrinsic rotation) is
`    q = exp((+roll/2)*e_x)*exp((+elev/2)*e_y)*exp((-azim/2)*e_z)`
i.e., Tait-Bryan angles, -z,y',x".


> The definitions in the docs are still unambiguous so I still think this is fairly marginal to make a change of API, but maybe @oscargus can weigh in.

Ah yes. So perhaps it _is_ time to open a separate Issue and close this PR ;)

**Pull request merged**
